### PR TITLE
parallel contact

### PR DESCRIPTION
**Pull request by SarahRo**
This PR introduces the necessary changes to the code to run two sided contact and meshties in parallel.
Some operations in parallel_mesh_ghosting.py do not scale in parallel. However, the functions provided in this file are necessary for running any examples in parallel. Improving upon this is future work.

### CODE REVIEW

File: python/dolfinx_contact/parallel_mesh_ghosting.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+from dolfinx.mesh import create_mesh, meshtags
+import dolfinx
+from dolfinx.cpp.mesh import entities_to_geometry, cell_num_vertices, cell_entity_type, to_type
+import numpy as np
+
+__all__ = ["create_contact_mesh"]
+
+
+def create_contact_mesh(mesh, fmarker, dmarker, tags):
```
Comment (jorgensd) [inline python/dolfinx_contact/parallel_mesh_ghosting.py:13]: I guess a long-term goal would be to move this function to C++, as iterating over number of cells can be costly

Comment (jorgensd) [inline python/dolfinx_contact/parallel_mesh_ghosting.py:13]: Some documentation of what happens to the ghosting of the mesh here would be useful.
Do we ghost all cells that are connected to facets of the opposite tag?

File: python/dolfinx_contact/parallel_mesh_ghosting.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    # Copy facets and markers to all processes
+    global_fmarkers = sum(fv_indices, [])
+    all_indices = mesh.comm.allgather(global_fmarkers)
+    all_indices = np.array(sum(all_indices, [])).reshape(-1, num_facet_vertices)
+    all_values = np.array(sum(mesh.comm.allgather(list(fmarker.values)), []))
+
+    global_dmarkers = sum(cv_indices, [])
+    all_cell_indices = mesh.comm.allgather(global_dmarkers)
+    all_cell_indices = np.array(sum(all_cell_indices, [])).reshape(-1, num_cell_vertices)
+    all_cell_values = np.array(sum(mesh.comm.allgather(list(dmarker.values)), []))
```
Comment (jorgensd) [inline python/dolfinx_contact/parallel_mesh_ghosting.py:44]: This seems really expensive.
Couldn't one use a neighborhood communicator based on if a process has any marked facets of a certain side or not?

Comment (SarahRo) [reply python/dolfinx_contact/parallel_mesh_ghosting.py:44]: You are right that this is really expensive. This function is only intended as a temporary fix to make thing run in parallel. There is no point in fixing individual problems if we are planning to completely replace this in a future pull request.

**Pull request merged**